src/pytesmo/scaling.py

```diff
@@ -568,3 +435,77 @@ def gen_cdf_match(
         scaled[scaled < min_val] = min_val
 
     return scaled
+
+
+@utils.deprecated()
+def cdf_beta_match(
+    src, ref, minobs=20, lin_edge_scaling=True, nbins=100, **kwargs
+):
+    """
+    takes the source timeseries, fits a beta distribution through its CDF and
+    finds unique percentile values corresponding to the number of bins used.
+    The size of bins is by default dynamically increased in case too few
+    observations (less than 20) are in a bin, leading to overfitting. Based on
+    Moesinger et al. (2020).
+
+    These values are used to scale the source to the reference by linear
+    interpolation between each pair of bin edges.
+
+    Uses the edge values linear scaling method described in Moesinger et
+    al. (2020) by default.
+
+    Parameters
+    ----------
+    src: numpy.array
+        input dataset which will be scaled
+    ref: numpy.array
+        src will be scaled to this dataset
+    minobs : int
+        Minimum desired number of observations in a bin.
+    nbins: int, optional
+        Number of bins to use for estimation of the CDF
+    ** kwargs: dict
+        keywords to be passed onto the gen_cdf_match() function
+
+    Returns
+    -------
+    CDF matched values: numpy.array
+        dataset src with CDF as ref
+    """
+    return cdf_match(src, ref, minobs=minobs, linear_edge_scaling=lin_edge_scaling, nbins=nbins)
+
+
+def cdf_match(
+    src, ref, minobs=20, linear_edge_scaling=True, nbins=100
+):
+    """
+    Rescales by CDF matching.
+
+    This calculates the empirical CDFs for source and reference dataset using a
+    specified number of bins. In case of non-unique percentile values, a beta
+    distribution is fitted to the CDF.
+    For more robust estimation of the lower and upper bins, linear edge scaling
+    is used (see Moesinger et al., 2020 for details).
+
+    Parameters
+    ----------
+    src: numpy.array
+        input dataset which will be scaled
+    ref: numpy.array
+        src will be scaled to this dataset
+    minobs : int
+        Minimum desired number of observations in a bin.
+    nbins: int, optional
+        Number of bins to use for estimation of the CDF
+    linear_edge_scaling : bool, optional
+        Whether to use linear edge scaling. Default is True.
+
+    Returns
+    -------
+    CDF matched values: numpy.array
+        dataset src with CDF as ref
+    """
+    matcher = CDFMatching(
```

Comment:
Shouldn't there be a keyword for `combine_invalid` of the `CDFMatching` class in the function here?

Reply:
Good idea, I added it.